backend/src/main/java/ca/gov/dtsstn/cdcp/api/service/SubscriptionService.java

```diff
@@ -24,6 +24,21 @@ public SubscriptionService(SubscriptionMapper mapper, SubscriptionRepository rep
 		this.repository = repository;
 	}
 
+	public Subscription create(Subscription subscription) {
+		Assert.notNull(subscription, "subscription is required; it must not be null");
+		Assert.isNull(subscription.getId(), "subscription.id must be null when creating new instance");
+
+		return mapper.fromEntity(repository.save(mapper.toEntity(subscription)));
+	}
+
+	public Subscription update(Subscription subscription) {
+		Assert.notNull(subscription, "subscription is required; it must not be null");
+		Assert.notNull(subscription.getId(), "subscription.id must not be null when updating existing instance");
+		final var originalPassportStatus = repository.findById(subscription.getId()).orElseThrow(); // NOSONAR (nullable param)
```

Comment:
**Nitpick:** you can remove the NOSONAR comment here, since we're not using SonarQube for this project.

Reply:
removed